pkg/util/containers/metrics/system/collector_linux.go

```diff
@@ -52,7 +52,7 @@ func newSystemCollector() (*systemCollector, error) {
 	var err error
 	var hostPrefix string
 
-	if !config.IsHostProcAvailable() || !config.IsHostSysAvailable() {
+	if !config.IsHostProcAvailable() || !config.IsHostSysAvailable() || !config.IsFeaturePresent(config.EKSFargate) {
```

Comment:
it think it is not exactly the fix that I had in mind. 

What we want is to not check the `host/proc` and `host/sys` availability in case eksFargate feature is present.
So more something like

```suggestion
	if !config.IsFeaturePresent(config.EKSFargate) && (!config.IsHostProcAvailable() || !config.IsHostSysAvailable())  {
```

Reply:
also we need to set the option `container_proc_root` to `/proc` instead of `host/proc` 
So either we document it in our doc. so user need to add another envvar (which is error prone), or we try to autoconfigure it base of the fact that `config.IsFeaturePresent(config.EKSFargate)`

Reply:
It looks like it might already be done? https://github.com/DataDog/datadog-agent/blob/feda0a42b88a93ceaafb31783bfb8c6f7a213a81/pkg/config/config.go#L411